Err cmplx format

yarondbb opened the pull request.

yarondbb committed:
Added check that backup_destination_folder structure is OK (array of Hash of 3 elements)

yarondbb committed:
updated error message

yarondbb committed:
Updated error message

yarondbb committed:
Updated error message

yarondbb committed:
Added check that monitoring_paths structure is OK (array of Hashes of 3 elements)

yarondbb committed:
updated error message

bbfsdev commented on lib/content_server/backup_server.rb:
```diff
@@ -42,6 +42,56 @@ def run_backup_server
     end
 
+    # check format of backup_destination_folder param to be array of hashes of 3 items
+    backup_destination_folder_struct_ok = true
```
1) Create several tests where you try to parse this parameter and succeed and fail in several cases.
2) Extract this to method is server.rb
3) Add check that the directories actually exist on drive.

yarondbb committed:
1. Moved check method to server.rb
2. Added tests

yarondbb committed:
removing params which are already defined

yarondbb committed:
Try yo fix stream issue for StringIO.new

yarondbb committed:
Fixed format

yarondbb committed:
Added Class for method call

yarondbb commented:
@kolmanv feature is ready for you to check

bbfsdev commented on spec/params/params_spec.rb:
```diff
@@ -80,6 +81,10 @@ module Spec
 
     describe 'Params::read_yml_params' do
+      # define dummy parameters for some tests below
+      #Params.complex('monitoring_paths', [{'path'=>'', 'scan_period'=>0, 'stable_state'=>0}], '')
```
Why are those two lines commented out?

yarondbb replied:
When they were valid, then rake shout that those params already been defined.. this is strange since there is no explicit require for file_monitoring.rb nor backup_server.rb.
I will investigate on this. Probably the way ruby runs the rake.

bbfsdev commented on spec/params/params_spec.rb:
```diff
@@ -172,6 +177,76 @@ module Spec
+      # expecting 1 path for backup_destination_folder
+      # (in call to check_monitoring_path_structure) but user gives
+      # 2 paths.
+      it 'should raise exception when backup_destination_folder format is bad 2' do
```
Change description to actually say what is the error, i.e.,
should raise exception when two backup_destination_folder defined, expect only one!

bbfsdev commented on spec/params/params_spec.rb:
```diff
@@ -172,6 +177,76 @@ module Spec
+      end
+
+      # missing 'stable_state' in hash
+      it 'should raise exception when monitoring_paths format is bad 3' do
```
Change description to:
should raise exception when missing stable_state parameter

bbfsdev commented on spec/params/params_spec.rb:
```diff
@@ -172,6 +177,76 @@ module Spec
+        expect {
+          Params.read_yml_params(StringIO.new(yml_good_format))
+          ContentServer.check_monitoring_path_structure('monitoring_paths', 0)
+        }.to_not raise_error
```
Please add expect to the data structure too, i.e., 
expect(Params['monitoring_path'][0].path).to eq('some_path_1')
expect(Params['monitoring_path'][0].scan_period).to eq(200)
...
expect(Params['monitoring_path'][1].path).to eq('some_path_2')
...

bbfsdev commented on lib/content_server/server.rb:
```diff
@@ -108,5 +108,51 @@ def flush_content_data
+      monitoring_paths_struct_ok = false
+    end
+
+    if not monitoring_paths_struct_ok
```
Extract this if as a function.

bbfsdev commented on lib/content_server/server.rb:
```diff
@@ -108,5 +108,51 @@ def flush_content_data
+    monitoring_paths_struct_ok = true
+    if structure.kind_of?(Array)
+      if (0 != expected_paths_size) and (expected_paths_size != structure.size)
+        monitoring_paths_struct_ok = false
```
You don't have to continue raise error here directly, use function you extract from below.

bbfsdev commented on lib/content_server/server.rb:
```diff
@@ -108,5 +108,51 @@ def flush_content_data
+      structure.each { |path_hash|
+        if path_hash.kind_of?(Hash)
+          if 3 != path_hash.size
+            monitoring_paths_struct_ok = false
```
Same for all places, raise function when you know the structure is not good.

kolmanv commented:
Done review.

On Thu, Aug 7, 2014 at 10:59 PM, yarondbb notifications@github.com wrote:

> @kolmanv https://github.com/kolmanv feature is ready for you to check
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/bbfsdev/bbfs/pull/271#issuecomment-51523560.

yarondbb committed:
code review following

yarondbb committed:
try fix param definition issue

yarondbb committed:
Added 'before' to test

yarondbb committed:
fixing parameters definitions
when running rake file as a stand alone then the condition inside the before is bot needed. If not (as we run it) it is needed

yarondbb committed:
typo

yarondbb committed:
removing redundent log

yarondbb committed:
separated conditions

yarondbb committed:
added has key method to params

yarondbb committed:
Merge pull request #274 from bbfsdev/zip_by_file_name

Zip by file name

yarondbb commented:
Manually merged to master
